File: go/day_two/day_two.go
package day_two

import (
    "bufio"
    "os"
    "strings"
)


func RockPaperScissors() (result int){
    theirchoices := map[string]int{
        "A": 0, // rock
        "B": 1, // paper
        "C": 2, // scissor
    }
    desired_results := map[string]int{
        "X": 0, // I lose
        "Y": 3, // draw
        "Z": 6, // I win
    }

    readPath := os.Args[1]
    file, err := os.Open(readPath)
    if err != nil {
        panic(err)
    }
    defer file.Close()

    fileScanner := bufio.NewScanner(file)

    fileScanner.Split(bufio.ScanLines)

    for fileScanner.Scan(){
        line := fileScanner.Text()

        choices := strings.Split(line, " ")
        their_choice := theirchoices[choices[0]]
        desired_result := desired_results[choices[1]]

        var my_choice int;
        switch desired_result{
        case 0:
            my_choice = (their_choice+2)%3
        case 3:
            my_choice = their_choice
        case 6:
            my_choice = (their_choice+1)%3
        }

        result += my_choice + desired_result + 1
    }

    return result
}

File: c_sharp/day_two.cs
namespace day_two{
    public static class day_two{
        private enum their_choices{A, B, C}
        private enum my_choices{X, Y, Z}

        public static int RockPaperScissors(string readPath){
            var result = 0;
            var lines = File.ReadLines(readPath);

            foreach(var line in lines){
                var choices = line.Split(" ");
                var their_choice = (int)Enum.Parse(typeof(their_choices), choices[0]);
                var my_choice = (int)Enum.Parse(typeof(my_choices), choices[1]);
                
                int result_score;
                if (my_choice == their_choice){
                    // draw
                    result_score = 3;
                }else if ((their_choice+1)%3 == my_choice){
                    // I win
                    result_score = 6;
                }else{
                    // I lose
                    result_score = 0;
                }

                result += result_score + my_choice + 1;
            }

            return result;
        }
    }
}
